provider/redis.go

```diff
@@ -34,58 +38,95 @@ func redisOnlineStoreFactory(serialized pc.SerializedConfig) (Provider, error) {
 	if redisConfig.Prefix == "" {
 		redisConfig.Prefix = "Featureform_table__"
 	}
-	return NewRedisOnlineStore(redisConfig), nil
+	return NewRedisOnlineStore(redisConfig)
 }
 
-func NewRedisOnlineStore(options *pc.RedisConfig) *redisOnlineStore {
-	redisOptions := &redis.Options{
-		Addr:     options.Addr,
-		Password: options.Password,
-		DB:       options.DB,
+func NewRedisOnlineStore(options *pc.RedisConfig) (*redisOnlineStore, error) {
+	redisOptions := rueidis.ClientOption{
+		InitAddress: []string{options.Addr},
+		Password:    options.Password,
+		SelectDB:    options.DB,
+	}
+	redisClient, err := rueidis.NewClient(redisOptions)
+	if err != nil {
+		return nil, err
 	}
-	redisClient := redis.NewClient(redisOptions)
 	return &redisOnlineStore{redisClient, options.Prefix, BaseProvider{
 		ProviderType:   pt.RedisOnline,
 		ProviderConfig: options.Serialized(),
 	},
-	}
+	}, nil
 }
 
 func (store *redisOnlineStore) AsOnlineStore() (OnlineStore, error) {
 	return store, nil
 }
 
+func (store *redisOnlineStore) AsVectorStore() (VectorStore, error) {
+	return store, nil
+}
+
 func (store *redisOnlineStore) Close() error {
-	// from the docs:
-	//
-	// "It is rare to Close a Client, as the Client is meant to be
-	// long-lived and shared between many goroutines.""
-	//
-	return store.client.Close()
+	store.client.Close()
+	return nil
 }
 
 func (store *redisOnlineStore) GetTable(feature, variant string) (OnlineStoreTable, error) {
 	key := redisTableKey{store.prefix, feature, variant}
-	vType, err := store.client.HGet(context.TODO(), fmt.Sprintf("%s__tables", store.prefix), key.String()).Result()
+	cmd := store.client.B().
+		Hget().
+		Key(fmt.Sprintf("%s__tables", store.prefix)).
+		Field(key.String()).
+		Build()
+	vType, err := store.client.Do(context.TODO(), cmd).ToString()
 	if err != nil {
 		return nil, &TableNotFound{feature, variant}
 	}
-	table := &redisOnlineTable{client: store.client, key: key, valueType: ValueType(vType)}
+	var valueType ValueType
+	if _, isScalarType := ScalarTypes[ScalarType(vType)]; isScalarType {
+		valueType = ScalarType(vType)
+	} else {
+		valueType = &VectorType{}
+		err = json.Unmarshal([]byte(vType), valueType)
+		if err != nil {
+			return nil, err
+		}
+	}
+	table := &redisOnlineIndex{
```

Comment:
You only return reidsOnlineIndex if its specifically and embedding otherwise you should return a redisOnlineTable. The reason being that normal tables don't have the Nearest method. It should also use the other Get and Set in that situation.

Reply:
It is funny that this works anyway but its not backward compatible